Feat/#300 dict diff filters

ciaranschutte opened the pull request:
- adds comparison filters (uses own `button` instead of `uikit/button` because overriding styles is proving rather obtuse)
- data hookup for filtering still to be implemented, will depends a lot on data needed for changes page, will talk to @kcullion tomorrow

hlminh2000 commented on website/src/pages/dictionary/index.js:
```diff
@@ -185,37 +196,63 @@ function DataDictionary() {
+              const { additionIsActive, deletionIsActive, updateIsActive } = compareFilters;
+
+              const comparisonBool =
+                (additionIsActive && compareVal === 'addition') ||
+                (deletionIsActive && compareVal === 'deletion') ||
+                (updateIsActive && compareVal === 'update');
```
there seems to be some scattered "language" that all seem to refer to the same things... `additionIsActive`, `addition` and `ADDITIONS` are all different manifestation of the same concept. Any chance we can reduce this variant? something like making `compareFilterTypes` something like `{ ADDITION: "ADDITION" ... }` maybe?

hlminh2000 commented on website/src/pages/dictionary/index.js:
```diff
@@ -185,37 +196,63 @@ function DataDictionary() {
+              }
+
+              const compareVal = get(field, 'compare', '');
+              const { additionIsActive, deletionIsActive, updateIsActive } = compareFilters;
```
`compareFilters` needs to be added to the `useMemo` dependency :)

hlminh2000 commented on website/src/pages/dictionary/index.js:
```diff
@@ -185,41 +196,67 @@ function DataDictionary() {
+             */
+            .map((fields, i) => {
+              if (i === 2) {
+                return { ...fields, ...{ compare: 'addition' } };
+              } else if (i === 3) {
+                return { ...fields, ...{ compare: 'deletion' } };
+              } else if (i === 7) {
+                return { ...fields, ...{ compare: 'update' } };
```
should we also just make this uppercase? maybe use `compareFilterTypes.ADDITION` for consistency

hlminh2000 commented on website/src/pages/dictionary/index.js:
```diff
@@ -185,41 +196,67 @@ function DataDictionary() {
+              const { ADDITION, DELETION, ACTIVE } = compareFilters;
+
+              const comparisonBool =
+                (ADDITION && compareVal === 'addition') ||
+                (DELETION && compareVal === 'deletion') ||
+                (ACTIVE && compareVal === 'update');
```
`compareFilterTypes.ADDITION`?

ciaranschutte commented:
@hlminh2000 fixed language

did some cleanup too

the major part being the `dictionary` and `diff` state.

We really need to figure out how we're getting the diffs because a lot of the UI depends on having a "diff type" key at the top of the data structure, we can talk more for next sprint. 

Really really really need to add TS to this project :D

ciaranschutte commented:
@hlminh2000 
- rebased to get latest docusaurus update to ensure it still builds/runs
- fixed build issue, component was in `pages` folder which is special folder for pages :(

Going ahead with merging this
